server/src/main/java/org/opensearch/action/search/SearchRequestStats.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.metrics.CounterMetric;
+import org.opensearch.common.metrics.MeanMetric;
+
+import java.util.function.Consumer;
+
+/**
+ * Request level search stats to track coordinator level node search latencies
+ *
+ * @opensearch.internal
+ */
+public final class SearchRequestStats implements SearchRequestOperationsListener {
```

Comment:
This class will need ongoing changes if/when we add new search phases.

It could mostly be reimplemented using an EnumMap where the keys are `SearchPhaseName`, and the values are a composite type of the `MeanMetric` and two `CounterMetric`s.